internal/dag/cache.go

```diff
@@ -105,6 +109,16 @@ func (kc *KubernetesCache) Insert(obj interface{}) bool {
 		}
 		kc.ingresses[m] = obj
 		return true
+	case *extensionsv1beta1.Ingress:
+		ingress := new(v1beta1.Ingress)
+		if err := translateIngress(obj, ingress); err != nil {
+			om := obj.GetObjectMeta()
+			kc.WithField("name", om.GetName()).
+				WithField("namespace", om.GetNamespace()).
+				Error(err)
+			return false
+		}
+		return kc.Insert(ingress)
```

Comment:
If the translate fails for some reason, should we log it? Also, in that case we'll insert an empty ingress. This not a stopper for this PR, just an observation.

Reply:
If translateIngress returns and error we log it then return false rather than falling through to kc.Insert.